gms/gossiper.cc

```diff
@@ -1860,10 +1860,7 @@ future<> gossiper::start_gossiping(int generation_nbr, std::map<application_stat
 
         auto generation = local_state.get_heart_beat_state().get_generation();
 
-        return replicate(get_broadcast_address(), local_state).then([] {
-            //notify snitches that Gossiper is about to start
-            return locator::i_endpoint_snitch::get_local_snitch_ptr()->gossiper_starting();
```

Comment:
One of the important things that is going to happen in the context of this call is a registration of a `reconnectable_helper` for snitches like GossipingPropertyFileSnitch and Ec2MultiDcSnitch. 

This callback is meant to enforce usage of 'local' addresses for inter-node RPC between nodes in the same DC.

In the original code it is registered before, for instance, a `_failure_detector_loop_done` assignment. And there are a few other `gossiper` state variables set here after the original call for a `gossiper_starting()`.

With the new code the `reconnect_helper` is going to be registered after all this.

@asias Could you, please, confirm that it's not going to break anything?

Reply:
@asias ping

Reply:
Will do.

Reply:
Btw. @avikivity  Let's delete test/manual/gossip.cc. It is not useful now. 

```

The triplication of the call is unfortunate. One duplication can be
removed by deleting test/manual/gossip.cc, which likely hasn't been
run in several years. The other duplication needs careful study of
storage_service startup.
```

Reply:
@avikivity I think you missed a place  to call 

locator::i_endpoint_snitch::get_local_snitch_ptr()->gossiper_starting().get()

in the bottom of storage_service::prepare_to_join where 

_gossiper.start_gossiping(generation_number, app_states, advertise).get();

is called.

Current patch does break things ;(

It is sad we need to call gossiper_starting everywhere. It is very easy to get it wrong. 

If you do want to reduce the dependency, why not just wrap the snitch start into a function object and pass it to gossiper.